pkgs/io_file/test/fifo.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'fifo_posix.dart';
+import 'fifo_windows.dart';
+
+/// An abstraction for a write-only object with a path.
+///
+/// Used to test reading from a file that does not report a reliable length.
+abstract class Fifo {
```

Comment:
I split "FifoWrite" from `read_as_bytes_test.dart` into an abstract class and separate POSIX and Windows implementations.